app/src/main/assets/native/externalplayer.js

```diff
@@ -130,9 +130,11 @@ define(['events', 'playbackManager', 'toast'], function (events, playbackManager
 
         self.notifyCanceled = function (message) {
             // required to not mark an item as seen / completed without time changes
-            var currentTime = self._currentTime || 0;
-            self.notifyTimeUpdate(currentTime + 1);
+            let currentTime = self._currentTime || 0;
             self.notifyTimeUpdate(currentTime - 1);
+            if (currentTime > 0) {
+                self.notifyTimeUpdate(currentTime);
+            }
```

Comment:
Doesn't this basically notify twice? Also, the first notify could push a `-1`, can't this cause issues?

Reply:
> Doesn't this basically notify twice? 

That's the point.
`// required to not mark an item as seen / completed without time changes`

Examples:

Case 1 - User start playback with media that has playerStartPositionTicks > 0 , external player not chosen
```
S1:E1  - playerStartPositionTicks : 60000
Android emits EVENT_CANCELED
currentTime = 6
notifyTimeUpdate(5);
notifyTimeUpdate(6);
notifyEnded()
Jellyfin report playback stopped by app
S1:E1  - playerStartPositionTicks: 60000
```

Case 2 - User start playback with media that has playerStartPositionTicks > 0 , external player not chosen, without time changes
```
S1:E1  - playerStartPositionTicks : 60000
Android emits EVENT_CANCELED
currentTime = 6
notifyEnded()
Jellyfin report playback stopped by app without time tracking, mark as watched
S1:E1  - playerStartPositionTicks: RunTimeTicks
```

Case 3 - User start playback with media that has playerStartPositionTicks > 0 , external player not chosen, with current time as time change
```
S1:E1  - playerStartPositionTicks : 60000
Android emits EVENT_CANCELED
currentTime = 6
notifyTimeUpdate(6);
notifyEnded()
Jellyfin report playback stopped by app without time tracking, mark as watched
S1:E1  - playerStartPositionTicks: RunTimeTicks
```

> Also, the first notify could push a `-1`, can't this cause issues?

Case 4 - User start playback with media that has playerStartPositionTicks = 0, external player not chosen
```
S1:E1  - playerStartPositionTicks: 0
Android emits EVENT_CANCELED
currentTime = 0
notifyTimeUpdate(-1);
notifyEnded()
Jellyfin server aborts playback, invalid time
S1:E1  - playerStartPositionTicks: 0
```

Case 5 - User start playback with media that has playerStartPositionTicks = 0, external player not chosen, without time changes
```
S1:E1  - playerStartPositionTicks : 0
Android emits EVENT_CANCELED
currentTime = 0
notifyEnded()
Jellyfin report playback stopped by app without time tracking, mark as watched
S1:E1  - playerStartPositionTicks: RunTimeTicks
```

Reply:
I see, thanks for the info. That makes sense then.